pvlib/iotools/solaranywhere.py

```diff
@@ -0,0 +1,292 @@
+    meta = results_json['WeatherDataResults'][0]['WeatherSourceInformation']
+    meta['time_resolution'] = results_json['WeatherDataResults'][0]['WeatherDataPeriods']['TimeResolution_Minutes']  # noqa: E501
+    meta['spatial_resolution'] = spatial_resolution
+    # Rename and convert applicable metadata parameters to floats
+    meta['latitude'] = float(meta.pop('Latitude'))
+    meta['longitude'] = float(meta.pop('Longitude'))
+    meta['altitude'] = float(meta.pop('Elevation_Meters'))
+    return data, meta
+
+
+def read_solaranywhere(filename, map_variables=True, encoding='iso-8859-1'):
+    """
+    Read a SolarAnywhere formatted file into a pandas DataFrame.
+
+    The SolarAnywhere file format and variables are described in [1]_. Note,
+    the SolarAnywhere file format resembles the TMY3 file format but contains
+    additional variables and meatadata.
+
+    Parameters
+    ----------
+    filename: str
+        Filename
+    map_variables: bool, default: True
+        When true, renames columns of the DataFrame to pvlib variable names
+        where applicable. See variable :const:`VARIABLE_MAP`.
+    encoding : str, default : 'iso-8859-1'
+        Encoding of the file. For SolarAnywhere TMY3 files the 'iso-8859-1'
+        encoding is recommended due to the usage of special characters.
+
+    Returns
+    -------
+    data: pandas.DataFrame
+        Timeseries data from SolarAnywhere.
+    metadata: dict
+        Metadata available in the file.
+
+    See Also
+    --------
+    pvlib.iotools.get_solaranywhere
+
+    References
+    ----------
+    .. [1] `SolarAnywhere historical data file formats
+       <https://www.solaranywhere.com/support/historical-data/file-formats/>`_
+    """
+    with open(str(filename), 'r', encoding=encoding) as fbuf:
+        # Extract first line of file which contains the metadata
+        firstline = fbuf.readline().strip().split(',')
+        # Read remaining part of file which contains the time series data
+        data = pd.read_csv(fbuf)
+
+    # Parse metadata
+    meta = {}
+    meta['USAF'] = int(firstline.pop(0))
+    meta['name'] = firstline.pop(0)
+    meta['state'] = firstline.pop(0)
+    meta['TZ'] = float(firstline.pop(0))
+    meta['latitude'] = float(firstline.pop(0))
+    meta['longitude'] = float(firstline.pop(0))
+    meta['altitude'] = float(firstline.pop(0))
+
+    # SolarAnywhere files contain additional metadata than the TMY3 format.
+    # The additional metadata is specified as key-value pairs, where each entry
+    # is separated by a slash, and the key-value pairs are separated by a
+    # colon. E.g., 'Data Version: 3.4 / Type: Typical Year / ...'
+    for i in ','.join(firstline).replace('"', '').split('/'):
+        if ':' in i:
+            k, v = i.split(':')
+            meta[k.strip()] = v.strip()
+
+    # Set index to UTC
+    data.index = pd.to_datetime(data['ObservationTime(GMT)'],
+                                format='%m/%d/%Y %H:%M', utc=True)
```

Comment:
```suggestion
    data.index = pd.to_datetime(data['ObservationTime(LST)'],
                                format='%m/%d/%Y %H:%M')
```
Maybe better to use the local standard time column?

Reply:
If so, I think the tests will need to be adjusted

Reply:
I think you are right that LST is preferred. My main point for saying this is that it's much easier for a user to convert to UTC than to convert to LST,